src/hooks/data/vaults/fallbacks/pricer/usePricer.ts

```diff
@@ -0,0 +1,131 @@
+import { Address } from "wagmi"
+
+import useAladdinApiPricer from "@/hooks/data/vaults/fallbacks/pricer/useAladdinApiPricer"
+import useApyVisionApiPricer from "@/hooks/data/vaults/fallbacks/pricer/useApyVisionApiPricer"
+import useCurveFactoryCryptoPricer from "@/hooks/data/vaults/fallbacks/pricer/useCurveFactoryCryptoPricer"
+import useCurveFactoryPricer from "@/hooks/data/vaults/fallbacks/pricer/useCurveFactoryPricer"
+import useCurveLpTokenPricer from "@/hooks/data/vaults/fallbacks/pricer/useCurveLpTokenPricer"
+import useCurveMainPricer from "@/hooks/data/vaults/fallbacks/pricer/useCurveMainPricer"
+import useGlpPricer from "@/hooks/data/vaults/fallbacks/pricer/useGlpPricer"
+import useLlamaApiPricer from "@/hooks/data/vaults/fallbacks/pricer/useLlamaApiPricer"
+import useLlamaArbiApiPricer from "@/hooks/data/vaults/fallbacks/pricer/useLlamaArbiApiPricer"
+import useLlamaEthPricer from "@/hooks/data/vaults/fallbacks/pricer/useLlamaEthPricer"
+
+import { PRIMARYASSET_PRICER } from "@/constant/mapping"
+
+export default function usePricer({
+  primaryAsset,
+  enabled,
+}: {
+  primaryAsset: Address | undefined
+  enabled: boolean
+}) {
+  const source = PRIMARYASSET_PRICER[primaryAsset ?? "0x"] ?? ""
+
+  const isEnabledLlamaApi = enabled && source === "LLAMA_API"
+  const isEnabledLlamaArbiApi = enabled && source === "LLAMA_ARBI_API"
+  const isEnabledAladdinApi = enabled && source === "ALADDIN_API"
+  const isEnabledApyVisionApi = enabled && source === "APY_VISION_API"
+  const isEnabledCurveFactory = enabled && source === "CURVE_FACTORY"
+  const isEnabledCurveFactoryCrypto =
+    enabled && source === "CURVE_FACTORY_CRYPTO"
+  const isEnabledCurveMain = enabled && source === "CURVE_MAIN"
+  const isEnabledLlamaEth = enabled && source === "LLAMA_ETH"
+  const isEnabledGlp = enabled && source === "GLP"
+  const isEnabledCurveLpToken = enabled && source === "CURVE_LPTOKEN"
+
+  const llamaApiQuery = useLlamaApiPricer({
```

Comment:
This works but I feel like it would be cleaner if you just returned one `useQuery` hook in here and changed the `queryFn` based on the enabled states. 

You just need to make sure you use a `queryKey` that will be unique for each enabled state.

Something like this...

```tsx
return useQuery({
  queryKey: ['pricer', source, primaryAsset],
  queryFn: source === "LLAMA" ? getLLama : source === "ALADDIN" ? getAlladin : ...
})
```

Then you can delete all of the hooks for each pricer and just export the getters. This will greatly reduce the amount of code here.

Reply:
done